src/Contexts/Authentication/AuthenticationIdentity.ts

```diff
@@ -0,0 +1,57 @@
+import IHttpWebToken from "../../Http/IHttpWebToken";
+import AuthenticationIdentityClaims from "./AuthenticationIdentityClaims";
+
+function tickToDate(tick: number): Date {
+    const d = new Date(0);
+    d.setUTCSeconds(tick);
+    return d;
+}
+
+function parseBearerToken(bearerToken: string): IHttpWebToken {
+    const payload: string = atob(bearerToken.split('.')[1]);
+    return JSON.parse(payload);
+}
+
+export default class AuthenticationIdentity {
+    public readonly bearerToken: string;
+    public readonly claims: AuthenticationIdentityClaims;
+
+    public constructor(bearerToken: string, claims: AuthenticationIdentityClaims) {
+        this.bearerToken = bearerToken;
+        this.claims = claims;
+    }
+
+    public static fromNoBearerToken() {
```

Comment:
I feel this is breaking some principle or another. If the bearer token is invalid your class should be able to infer that.